src/detect-xbits.c

```diff
@@ -271,6 +275,33 @@ static int DetectXbitParse(DetectEngineCtx *de_ctx,
                 }
                 SCLogDebug("expire %d", expire);
             }
+            int32_t vsize =  pcre2_get_ovector_count(parse_regex.match); // Get the number of pairs of values
+            int32_t stripped_rawstr_len = rawstr_len;
+            /* Calculate the length of raw string without chars parsed or returned
+             * by the pcre lib */
+            for (int i=0; i < rawstr_len; i++) {
+                if (rawstr[i] == ' ' || rawstr[i] == ',') {
+                    stripped_rawstr_len -= 1;
+                }
+            }
+            const char *bufptr, *prev_buf;
+            for (int i = 1; i <= vsize; i++) {
+                size_t buflen = 0;
+                res = pcre2_substring_get_bynumber(parse_regex.match, i, (PCRE2_UCHAR **)&bufptr, (PCRE2_SIZE *)&buflen);
+                if (res < 0 || bufptr == NULL) {
+                    continue; // We know that ovector may have more values of interest for us
+                }
+                /* Avoid counting length of repeated buffers as ovector can keep
+                 * up to 2n+1 values, where n = no. of capturing parentheses */
+                if (bufptr != prev_buf) { // FIXME Dangerous check? Could it be wrong in any cases?
+                    args_len += strlen(bufptr);
+                }
+                prev_buf = bufptr;
+            }
+            if (args_len != stripped_rawstr_len) {
+                SCLogError(SC_ERR_INVALID_VALUE, "Invalid values found in the xbits setting");
+                return -1;
+            }
```

Comment:
This is probably too long and seemingly dirty way to achieve the fix. Better suggestions?